### ISSUE TITLE
Add signed source tar.gz release packages

### ISSUE THREAD

**Issue by ehdis** [opentofu/opentofu#2107]
### OpenTofu Version

```shell
All
```

### The problem in your OpenTofu project

In addition to the compiled release packages, it would be great to have also **signed** _source_ tar.gz archives (obviously not the dynamical generated). 

### Attempted Solutions

Every compiled release package can be integrity-checked/validated but not the sources! 

### Proposal

Extend the action workflow to build and sign also a source tarball.

**Comment by ehdis** [comment]
not necessarily. Before building the project code on a local system, its important to check the integrity of the sources of the actual project (this request). Dependencies can come from the OS (then already integrity-checked) or as self-contained vendor directory created locally under best-pratices to address supply-chain-security, but this is a separate issue. When the project includes a vendor directory by it self, then it should still offer a signed source package version.

**Comment by Yantrio** [comment]
Thanks for raising this issue. GitHub automatically generates the .tar.gz files for each release, ensuring consistency and correctness. If we manually produce an additional signed .tar.gz file, it introduces potential for human error and increases complexity.

Given this, I believe it's preferable to rely on GitHub's built-in validation process and not introduce an additional signed .tar.gz asset. Users should trust the integrity provided by GitHub's existing mechanisms for release artifacts.

I'll close this issue but please feel free to carry on any discussion and the core team can revisit this later.
